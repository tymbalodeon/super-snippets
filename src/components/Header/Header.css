.Header {
  text-align: center;
}

.Header__not-logged-in > a,
.Header__logged-in > a {
  color: #d4bfff;
}

.Header__not-logged-in > a {
  padding: 10px;
}

.super {
  color: #ffa759;
}

.snippets {
  color: #f28779;
}

.brackets {
  color: #cbccc6;
  font-size: 36px;
}
