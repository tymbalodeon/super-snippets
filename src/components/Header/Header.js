import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Header.css';

class Header extends Component {
  handleLogoutClick = () => {
    const { history } = this.props;
    localStorage.removeItem('super-snippets-client-auth-token');
    history.push('/');
  };

  renderLogoutLink() {
    return (
      <div className="Header__logged-in">
        <Link onClick={this.handleLogoutClick} to="/">
          Logout
        </Link>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div className="Header__not-logged-in">
        <Link to="/login">Log in</Link>
        <Link to="/register">Register</Link>
      </div>
    );
  }

  render() {
    const { updateProjectId } = this.props;
    const loggedIn = localStorage.getItem('super-snippets-client-auth-token');
    return (
      <>
        <nav className="Header">
          <h1>
            <Link to="/snippets" onClick={() => updateProjectId()}>
              <span className="brackets">{'{ '}</span>
              <span className="super">Super</span>
              <span className="snippets">Snippets</span>
              <span className="brackets">{' }'}</span>
            </Link>
          </h1>
          <span className="Header__tagline--wide">Collide all the things.</span>
          {loggedIn ? this.renderLogoutLink() : this.renderLoginLink()}
        </nav>
      </>
    );
  }
}

export default Header;
