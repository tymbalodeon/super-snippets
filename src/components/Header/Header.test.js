import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('<Header />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
