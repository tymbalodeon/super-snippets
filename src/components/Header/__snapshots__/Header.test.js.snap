// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the UI as expected 1`] = `
<nav
  className="Header"
>
  <h1>
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="brackets"
      >
        { 
      </span>
      <span
        className="super"
      >
        Super
      </span>
      <span
        className="snippets"
      >
        Snippets
      </span>
      <span
        className="brackets"
      >
         }
      </span>
    </a>
  </h1>
  <div
    className="Header__not-logged-in"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      Log in
    </a>
    <a
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</nav>
`;
